File: 영어끝말잇기.py
def solution(n, words):
    past = []
    ans = [0,0]
    past.append(words[0])
    
    for i in range(1, len(words)) :
        ans1 = i // n + 1
        ans2 = i % n + 1
        if words[i-1][-1] != words[i][0] :
            return [ans2,ans1]
        elif words[i] in past:
            return [ans2,ans1]
        
        past.append(words[i])
        
    return ans

File: 영어끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    vector<int> answer;
   
   	for (int i = 1; i < words.size(); i++) {
		int num = i%n + 1 ;
   		int turn = i/n + 1;

   		for (int j = 0; j < i; j++) {
   			if (words[i] == words[j]) {
   				answer = {num, turn};
   				return answer;
   			}
   		}

   		int size = words[i-1].size();
   		
   		if (words[i-1][size-1] != words[i][0]) {
   			answer = {num, turn};
   			return answer;

   		} 

   	}

   	answer = {0,0};

    return answer;
}